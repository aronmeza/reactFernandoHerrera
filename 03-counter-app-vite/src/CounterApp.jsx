import { useState } from 'react'
import PropTypes from "prop-types";

export const CounterApp = ({value}) =>{

  const [ counter, setCounter ] = useState(value)
  const handleAdd= ()=>{
    // setCounter(counter + 1);
    setCounter((c) => c+1)
  }
  const handleRemove= ()=>{ setCounter((c) => c-1) }
  const handleReseat= ()=>{ setCounter((c) => value) }
  return(
    <>
      <h1>Counter App</h1>
      <h2> {counter} </h2>
      <button onClick={  handleAdd }>+1</button>
      <button onClick={  handleRemove }>-1</button>
      <button aria-label="btn-reseat" onClick={  handleReseat }>Reseat</button>
    </>
  );
}


CounterApp.propType = {
  value: PropTypes.number
}
