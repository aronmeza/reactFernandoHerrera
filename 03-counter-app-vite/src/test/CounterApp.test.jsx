import {fireEvent, render, screen} from "@testing-library/react";
import {CounterApp} from "../CounterApp.jsx";

describe('Pruebas en main de <CounterApp/>', () =>{
  let contadorInicial = 100;
  test('Debe hacer match con el snapchat', () =>{
    const { container } = render(<CounterApp value={contadorInicial}/>)
    expect(container).toMatchSnapshot();

  });

  test('mostrar el valor inicial en 100', () =>{
    render(<CounterApp value={contadorInicial}/>)
    expect(screen.getByText(contadorInicial)).toBeTruthy();
    //expect(screen.getByRole('heading',{level:2}).innerHTML).toContain('100')
  });

  test('debe de incrementear con el boton +1', () =>{
    render(<CounterApp value={contadorInicial}/>)
    fireEvent.click(screen.getByText('+1'))
    expect(screen.getByText('101')).toBeTruthy();

  });

  test('debe de decrementear con el boton -1', () =>{
    render(<CounterApp value={contadorInicial}/>)
    fireEvent.click(screen.getByText('-1'))
    expect(screen.getByText('99')).toBeTruthy();

  });

  test('debe resetear el valor inicial', () =>{
    render(<CounterApp value={contadorInicial}/>)
    fireEvent.click(screen.getByText('+1'))
    fireEvent.click(screen.getByText('+1'))
    fireEvent.click(screen.getByText('+1'))
    expect(screen.getByText('103')).toBeTruthy();
    //fireEvent.click(screen.getByText('Reseat'))
    fireEvent.click(screen.getByRole('button',{name:'btn-reseat'}))
    expect(screen.getByText('100')).toBeTruthy();

  });

});
