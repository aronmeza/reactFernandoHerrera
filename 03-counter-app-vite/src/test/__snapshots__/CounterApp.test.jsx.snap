// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en main de <CounterApp/> Debe hacer match con el snapchat 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reseat
  </button>
</div>
`;
